### PR TITLE
Force storing of StreamerInfo for nested non-split but memberwise streamed collection.

### LINKED ISSUE #11436: Missing StreamerInfo in file containing nested collection that is non-split but stored member wise
With example below, the resulting file is missing the StreamerInfo for the `Inside` class preventing schema evolution for this class. 

```C++
#include <vector>
#include "TTree.h"
#include "TFile.h"

struct Inside {
  int fValue;
};

struct Middle {
  std::vector<Inside> fInside;
};

struct Outer {
  std::vector<Middle> fMiddle;
};

void writefile(bool empty = true, const char *filename = "forceinfo.root")
{
   TFile *f = TFile::Open(filename, "RECREATE");
   TTree *t = new TTree("t", "t");
   Outer object;
   t->Branch("object.", &object);

   if (!empty) {
      Inside in;
      Middle m;
      m.fInside.push_back(in);
      object.fMiddle.push_back(m);
   }

   t->Fill();
   f->Write();
   delete f;
}
```

### PR DESCRIPTION
**Pull request by pcanal**
This fix #11436

Corresponding roottest PR: https://github.com/root-project/roottest/pull/906

### CODE REVIEW

File: io/io/src/TStreamerInfo.cxx [hunk 1]
```diff
@@ -3197,6 +3215,9 @@ void TStreamerInfo::ForceWriteInfo(TFile* file, Bool_t force)
          return;
       }
    } else if (fClass->GetCollectionProxy()) { // We are an STL collection.
+      TClass *valueClass = fClass->GetCollectionProxy()->GetValueClass();
+      if (false && valueClass)
```
Comment (Axel-Naumann) [inline io/io/src/TStreamerInfo.cxx:3219]:
```suggestion
      if (TClass *valueClass = fClass->GetCollectionProxy()->GetValueClass())
```

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]
Just making sure you saw the crucial functional change that's required...

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
All test passes as reported my the related roottest PR: https://github.com/root-project/roottest/pull/906#issuecomment-1265687117.

**Pull request merged**